app/lib/env.server.ts

```diff
@@ -44,6 +44,22 @@ function getHostname() {
   return new URL(getOrigin()).hostname
 }
 
+function getDomain() {
+  const hostname = getHostname() as
+    | 'gcn.nasa.gov'
+    | 'test.gcn.nasa.gov'
+    | 'dev.gcn.nasa.gov'
+    | undefined
+
+  if (hostname === 'gcn.nasa.gov') {
+    return undefined
+  } else if (hostname?.endsWith('gcn.nasa.gov')) {
+    return hostname
+  } else {
+    return 'dev.gcn.nasa.gov'
```

Comment:
Please keep the default set to the public testing server. An external contributor would have access to that but not to dev.gcn.nasa.gov.

```suggestion
    return 'test.gcn.nasa.gov'
```